kbfsgit/runner.go

```diff
@@ -579,62 +596,127 @@ func humanizeObjects(n int, d int) string {
 	return fmt.Sprintf("%.2f/%.2fM", float64(n)/m, float64(d)/m)
 }
 
-func (r *runner) processGogitStatus(
-	ctx context.Context, statusChan <-chan plumbing.StatusUpdate) {
+func (r *runner) printJournalStatusUntilFlushed(
+	ctx context.Context, doneCh <-chan struct{}) {
+	rootNode, _, err := r.config.KBFSOps().GetOrCreateRootNode(
+		ctx, r.h, libkbfs.MasterBranch)
+	if err != nil {
+		r.log.CDebugf(ctx, "GetOrCreateRootNode error: %+v", err)
+		return
+	}
+
+	err = r.config.KBFSOps().SyncAll(ctx, rootNode.GetFolderBranch())
+	if err != nil {
+		r.log.CDebugf(ctx, "SyncAll error: %+v", err)
+		return
+	}
+
+	jServer, err := libkbfs.GetJournalServer(r.config)
+	if err != nil {
+		r.log.CDebugf(ctx, "No journal server: %+v", err)
+	}
+
+	r.printJournalStatus(
+		ctx, jServer, rootNode.GetFolderBranch().Tlf, doneCh)
+}
+
+func (r *runner) processGogitStatus(ctx context.Context,
+	statusChan <-chan plumbing.StatusUpdate, fsEvents <-chan libfs.FSEvent) {
 	currStage := plumbing.StatusUnknown
 	var startTime time.Time
 	lastByteCount := 0
 	cpuProf := ""
-	for update := range statusChan {
-		if update.Stage != currStage {
-			if currStage != plumbing.StatusUnknown {
-				memProf := fmt.Sprintf("mem.%d.prof", currStage)
-				elapsedStr := r.getElapsedStr(ctx, startTime, memProf, cpuProf)
-				r.errput.Write([]byte("done." + elapsedStr + "\n"))
+	donePrinted := true
+	for {
+		if fsEvents == nil && statusChan == nil {
+			// Both channels are closed.
```

Comment:
`nil` is different from closed, right?

Reply:
They are set to `nil` when closed (in each case below). I guess I should add some more comments.

Reply:
Oh I see, I thought the comment was referring to the channels as they were passed in.